File: SensorMod/src/main/java/com/programminghoch10/SensorMod/Hook.java
package com.programminghoch10.SensorMod;

import static com.programminghoch10.SensorMod.Common.DEFAULT_SENSOR_STATE;
import static com.programminghoch10.SensorMod.Common.getKey;

import android.content.SharedPreferences;
import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.util.Log;

import java.util.List;

import de.robv.android.xposed.IXposedHookLoadPackage;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XSharedPreferences;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class Hook implements IXposedHookLoadPackage {
    
    @Override
    public void handleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) throws Throwable {
        if (lpparam.packageName.equals(BuildConfig.APPLICATION_ID)) return;
        SharedPreferences sharedPreferences = new XSharedPreferences(BuildConfig.APPLICATION_ID, "sensors");
        Class<SensorManager> systemSensorManager =
            (Class<SensorManager>) XposedHelpers.findClass("android.hardware.SystemSensorManager", lpparam.classLoader);
        XposedHelpers.findAndHookMethod(
            systemSensorManager, "getFullSensorList", new XC_MethodHook() {
                @Override
                protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                    if (param.hasThrowable()) return;
                    Log.d("Logger", "afterHookedMethod: filter getFullSensorList");
                    List<Sensor> result = (List<Sensor>) param.getResult();
                    result = result.stream().filter(sensor -> sharedPreferences.getBoolean(getKey(sensor), DEFAULT_SENSOR_STATE)).toList();
                    param.setResult(result);
                }
            }
        );
        Log.d("Logger", "handleLoadPackage: Hooked SystemSensorManager");
    }
}

File: KeepSplitScreenRatio/src/main/java/com/programminghoch10/KeepSplitScreenRatio/Hook.kt
package com.programminghoch10.KeepSplitScreenRatio

import android.os.IBinder
import de.robv.android.xposed.IXposedHookLoadPackage
import de.robv.android.xposed.XposedHelpers
import de.robv.android.xposed.callbacks.XC_LoadPackage
import de.robv.android.xposed.XC_MethodHook as MethodHook

class Hook : IXposedHookLoadPackage {
    
    override fun handleLoadPackage(lpparam: XC_LoadPackage.LoadPackageParam) {
        if (lpparam.packageName != "com.android.systemui") return
        
        val SplitScreenTransitionsClass = XposedHelpers.findClass("com.android.wm.shell.splitscreen.SplitScreenTransitions", lpparam.classLoader)
        XposedHelpers.findAndHookMethod(
            SplitScreenTransitionsClass,
            "setEnterTransition",
            IBinder::class.java,
            "android.window.RemoteTransition",
            Int::class.java,
            Boolean::class.java,
            object : MethodHook() {
                override fun beforeHookedMethod(param: MethodHookParam) {
                    param.args[3] = false
                }
            },
        )
    }
}
